### PR TITLE
Handles creating element areas for each page that extends elemental p…

### PR DESCRIPTION
**Pull request by StephenMakrogianni**
…age extension.

Sets Subsite::$disable_subsite_filter to true, reruns ElementalAreasExtension's requireDefaultRecords() then sets Subsite::$disable_subsite_filter to it's original state

### CODE REVIEW

File: src/Extensions/ElementalSubsitesPageExtension.php [hunk 1]
```diff
@@ -45,4 +48,24 @@ public function onAfterDuplicateToSubsite($originalPage)
...
+     * ElementalAreasExtension's requireDefaultRecords() then
+     * Sets Subsite::$disable_subsite_filter to it's original state
+     *
+     * @return void
+     */
+    public function requireDefaultRecords()
+    {
+        $original = Subsite::$disable_subsite_filter;
+        Subsite::$disable_subsite_filter = true;
+        $instance = Injector::inst()->get(ElementalAreasExtension::class);
```
Comment (robbieaverill) [inline src/Extensions/ElementalSubsitesPageExtension.php:65]: Wouldn't this logic mean that `ElementalAreasExtension::requireDefaultRecords()` ends up running twice?

Comment (robbieaverill) [reply src/Extensions/ElementalSubsitesPageExtension.php:65]: (Friendly) ping =)

Comment (StephenMakrogianni) [reply src/Extensions/ElementalSubsitesPageExtension.php]: From my understanding this would override that function and run only here.

Comment (StephenMakrogianni) [reply src/Extensions/ElementalSubsitesPageExtension.php:65]: Yeah you're right. Instead i'll use a hook.

Comment (StephenMakrogianni) [reply src/Extensions/ElementalSubsitesPageExtension.php:65]: @robbieaverill I have closed thin in favor of this PR: https://github.com/dnadesign/silverstripe-elemental-subsites/pull/10

**Pull request closed without merging**